cypress_test/integration_tests/common/impress_helper.js

```diff
@@ -124,7 +124,7 @@ function removeShapeSelection() {
 
 		return cy.cGet('.leaflet-overlay-pane svg')
 		.then(function(overlay) {
-				return overlay.children('g').length === 0;
```

Comment:
This change doesn't seem reliable to me. Shape selection may persist after the click and "cursor" won't exist. This will return true if the shape is selected + its cursor is not visible.

Checking 'g' element is more reliable here. Because it exists if the shape is selected.

If we want to distinguish the "shape selected" and "shape text being edited" cases, i think we need another function. Or if this is used only by one place, then we can rename this to something like "cancel shape text edit" etc etc.

Reply:
@lpranam and I have both been struggling with this. There is sometimes an extra g element that means the check was never returning true. This was the only reliable query I could find.

What this really needs is some changes to the DOM to make shape selection/text selection/no selection more detectable, then updates to the tests to use a new simpler query. That's too much to do here in this change right now.

Reply:
Let me also check this, i am wondering about the extra g element. Maybe we can find a formula for that. We have time for now, since the tests haven't passed yet.

Reply:
I think there are 2 scenarios for this:
* Text edit is enabled on the selected shape. A click to somewhere else disables the text edit but doesn't de-select the shape.
* A click to somewhere else selects another shape.
* These 2 scenarios would still show a g element in the div.
* If text selection is not disabled, then there are 2 g elements. But i think the extra g element you saw was due to selecting another shape.

I think we can use:
* cy.cGet('body').type('{esc}'); -> To disable text editing on shape.
* Then one more:
* cy.cGet('body').type('{esc}'); -> To deselect the shape.

This should be safe and shouldn't cause a leftover element.

Reply:
Did you try this? I couldn't get the escape keypress to work.

Also I don't think it's from selecting another shape. As far as I can tell there's only one shape and the click is in empty space outside the slide anyway.

Reply:
We can merge this and you may prefer a follow up for this if you wish? Since this passed CI. Maybe a follow up will be more comfortable?

Reply:
Yeah let's merge this to get mobile tests running for all the other CI jobs. Then I'll try this.

Reply:
Okay :)